### PR TITLE
pscanrules: Add Example Alerts to CrossDomainScriptInclusionScanRule

### PR DESCRIPTION
**Pull request by andremralves**
Part of https://github.com/zaproxy/zaproxy/issues/6119

### Changes:
- Add example alert functionality.
- Add unit tests for example alerts functionality.
- Add note to changelog.

### CODE REVIEW

File: addOns/pscanrules/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
+### Changed
+- The Cross-Domain Javascript Source File Inclusion scan rule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:8]: This should be moved to existing Changed section.

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java [hunk 2]
```diff
@@ -84,17 +84,28 @@ && isScriptFromOtherDomain(
...
+
+    private void raiseAlert(HttpMessage msg, int id, String crossDomainScript, String evidence) {
+        createAlert(crossDomainScript, evidence).raise();
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(createAlert(
+                "<script type=\"text/javascript\" src=\"externalDomain.example.com/weatherwidget.js\"></script>",
+                null
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java:107]: I believe the parameters are swapped.

Comment (andremralves) [reply addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java:107]: You are right, thank you! The `evidence` variable is the whole script tag and the `crossDomainScript` is just the `src` attribute, right?

**Comment by andremralves** [comment]
> This needs a `spotlessApply`.

I was trying to find a formatting command and didn't know about this one, thanks.

**Comment by andremralves** [comment]
I made the requested changes.

File: addOns/pscanrules/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 - Application Error Scan Rule no longer checks JavaScript or CSS responses unless threshold is Low (Issue 7724).
+- The Cross-Domain Javascript Source File Inclusion scan rule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:13]: `JavaScript`

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
How would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Comment by andremralves** [comment]
> How would you like to be credited (e.g. name, handle)? https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

André Alves ([@andrealvesdev](https://twitter.com/andrealvesdev))

Thank you for adding me, I hope to contribute with more issues soon.

**Pull request merged**